packages/analytics-js/src/services/HttpClient/xhr/xhrRequestHandler.ts

```diff
@@ -80,7 +80,9 @@ const xhrRequest = (
     xhr.open(options.method, options.url);
 
     Object.keys(options.headers).forEach(headerName => {
```

Comment:
Headers are optional. We should validate for that.

Reply:
headers is not optional, if none were passes sensible defaults are merged with them DEFAULT_XHR_REQUEST_OPTIONS

Reply:
Makes sense. I'm talking about the input headers to this client.